src/event.rs

```diff
@@ -250,34 +253,42 @@ impl<N: Notify> Processor<N> {
                 }
             }
 
-            match window.wait_events().next() {
+            let event = if self.wait_for_event {
+                window.wait_events().next()
+            } else {
+                None
+            };
+
+            terminal = term.lock();
+
+            context = ActionContext {
+                terminal: &mut terminal,
+                notifier: &mut self.notifier,
+                selection: &mut self.selection,
+                mouse: &mut self.mouse,
+                size_info: &self.size_info,
+            };
+
+            processor = input::Processor {
+                ctx: context,
+                key_bindings: &self.key_bindings[..],
+                mouse_bindings: &self.mouse_bindings[..]
+            };
+
+            match event {
                 Some(event) => {
-                    terminal = term.lock();
-                    context = ActionContext {
-                        terminal: &mut terminal,
-                        notifier: &mut self.notifier,
-                        selection: &mut self.selection,
-                        mouse: &mut self.mouse,
-                        size_info: &self.size_info,
-                    };
-
-                    processor = input::Processor {
-                        ctx: context,
-                        key_bindings: &self.key_bindings[..],
-                        mouse_bindings: &self.mouse_bindings[..]
-                    };
-
                     process!(event);
                 },
-                // Glutin guarantees the WaitEventsIterator never returns None.
-                None => unreachable!(),
+                None => ()
             }
 
             for event in window.poll_events() {
                 process!(event);
             }
         }
 
+        self.wait_for_event = !terminal.dirty;
```

Comment:
I think we can just put it in an issue. The issue is that 1 extra frame will be drawn whether it's needed or not. The locking of `Term` is going to be annoying to resolve it.

We should probably try and merge this branch and file an issue as you suggested.